### ISSUE TITLE
Premium and Cracked Accounts

### ISSUE THREAD

**Issue by zynwire** [SilkePilon/KnowledgeBook#46]
First off, can you use your Premium (Actual) Accounts from Minecraft? Or is it only cracked, since I couldn't start it yet because I get this error running under Fedora 
![image](https://github.com/user-attachments/assets/7dfdc190-9b79-41e8-b30e-2505514b6259)
I'll just try Ubuntu, I'll say everything that happens. Also, great work. This is crazy stuff and it's really cool.

**Comment by SilkePilon** [comment]
"getaddrinfo ENOTFOUND means client was not able to connect to given address." The backend will try to connect and run its server on your local network ip. (for example 192.168.X.X) but it seems its unable to.

i will add an option to select and ip!

give me a few seconds and you will need to delete the dir and rerun the command.

also funny thing: this project only works with paid (Premium) accounts. once you enter the username it asks you to login. only if you set the host to localhost it will use an offline account. this is to stop users spamming servers.

i will comment below once i update the code (should be a few mintues)

**Comment by zynwire** [comment]
Perfect timing, just finished installing Ubuntu. Tell me when you updated it and I'll install it fresh.

**Comment by SilkePilon** [comment]
added!

**Comment by SilkePilon** [comment]
it may take a few seconds to update over on github's servers tho

**Comment by zynwire** [comment]
Yup, I'll install it in a sec

**Comment by SilkePilon** [comment]
let me know if you have any issues!

**Comment by zynwire** [comment]
Please wait...
node:internal/modules/cjs/loader:1137
  throw err;
  ^

Error: Cannot find module 'mineflayer'
Require stack:
- /home/erik/OpenDeliveryBot/main.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1134:15)
    at Module._load (node:internal/modules/cjs/loader:975:27)
    at Module.require (node:internal/modules/cjs/loader:1225:19)
    at require (node:internal/modules/helpers:177:18)
    at Object.<anonymous> (/home/erik/OpenDeliveryBot/main.js:3:20)
    at Module._compile (node:internal/modules/cjs/loader:1356:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1414:10)
    at Module.load (node:internal/modules/cjs/loader:1197:32)
    at Module._load (node:internal/modules/cjs/loader:1013:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:128:12) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [ '/home/erik/OpenDeliveryBot/main.js' ]
}

Node.js v18.19.1
Tried to run node main.js

**Comment by zynwire** [comment]
Nvm, I'm stupid.

**Comment by zynwire** [comment]
Nvm installing the module didnt work

**Comment by SilkePilon** [comment]
Mmmm this seems to be more a node issue than my code issue

**Comment by SilkePilon** [comment]
Try running npm install

**Comment by SilkePilon** [comment]
Or look up: error ubuntu npm cant find packages

**Comment by zynwire** [comment]
Npm install just installes the same package, doesn't fix it really. Idk I guess Mineflyer doesnt work

**Comment by SilkePilon** [comment]
Mineflyer is the 1 package that is needed. tho i know it can be installed

**Comment by SilkePilon** [comment]
npm i should install it

**Comment by SilkePilon** [comment]
it works fine on my windows (11) pc and on the github codespaces (linux with node 16 i think)

**Comment by SilkePilon** [comment]
maybe:
```
sudo rm -rf /usr/local/lib/node_modules/npm 
```
and then re-install Node.js will work in most cases

from https://stackoverflow.com/a/47083659

**Comment by zynwire** [comment]
I'll do that after school tomorrow.

**Comment by zynwire** [comment]
Also, is the Nvidia API key required?

**Comment by SilkePilon** [comment]
No, tho it's needed if you want to generate nodes using ai, the key is free to get and only requires an account. You can make 1000 nodes using it. There are instructions on the site on how to obtain a key if you want one

**Comment by SilkePilon** [comment]
i also just changed the url to https://knowledgebook.vercel.app/

**Comment by zynwire** [comment]
Is there an AutoLog feature? Like when you get hit the bot disconnects?

**Comment by zynwire** [comment]
Going to school in abit, I'll try the thing around 12:30

**Comment by SilkePilon** [comment]
Well you can make the ai code an autolog node.

Or if you want I can make it an switch in the dashboard

**Comment by zynwire** [comment]
It would be better to be a switch in the Dashboard

**Comment by zynwire** [comment]
Got it working

**Comment by zynwire** [comment]
Just creating the bot since my machine is slow which I'm running this on

**Comment by zynwire** [comment]
Okay, so the thing generally works, but a few things to mention. So firstly it's really slow, but that's the issue of my specs. But then when I try to connect using another PC I can't it's the same network and stuff but still.

**Comment by SilkePilon** [comment]
some schools bock things like this. is this at school or home.
if you want to access it on other pc's in the same network make sure to use the first option from the list when starting the backend. if its running on localhost it wont work as that can only be accessed on the pc its running on.

**Comment by zynwire** [comment]
Nah im at home

**Comment by zynwire** [comment]
Is it possible to grab Planks from my Ender Chest and deliver them to a spot where there is a normal chest?

**Comment by SilkePilon** [comment]
yes. atm you do need to make it using nodes. so make a node that gets items from the ender chest, then a node that walks to a location and then a node that drops off the items. You can do all this uisng the ai option

**Comment by zynwire** [comment]
Does the ai work well?

**Comment by SilkePilon** [comment]
yea and its free

**Comment by SilkePilon** [comment]
@ErikLavr do you have any suggestions of what I can add next?

I don't get a lot of feedback so it would be really appreciated!

**Comment by zynwire** [comment]
Prob auto log. That would make it perfect. For 6b6t a TPA mode maybe?

**Comment by SilkePilon** [comment]
Sure i can add that!

**Comment by SilkePilon** [comment]
@ErikLavr added!

**Comment by zynwire** [comment]
Tysm, didn't see your message in time.

**Comment by zynwire** [comment]
Hey so a little question, I wanted to make my own Mineflayer bot that uses Mineflayer-elytrafly thingy but i can get it to work. I would like to ask you for help since youre the best one to ask. Chat GPT doesnt really help either the docs arent really good and yeah. Its on 8b8t. I want it to just fly around and search for stashes. Like random positions. If it finds one it Logs it to a TxT file. I really have tried everything (and also taking some stuff from your code) but it just doesnt work. Could you help me?

**Comment by SilkePilon** [comment]
Sorry for not responding to your message @zynwire did you get it working?
